pyccel/ast/cupyext.py

```diff
@@ -63,14 +63,14 @@ class CupyNewArray(CudaNewArray):
     """ Class from which all Cupy functions which imply a call to Allocate
     inherit
     """
-    __slots__ = ('_arg','_dtype','_precision','_shape','_rank','_order', '_memory_location')
+    __slots__ = ('_memory_location')
     def __init__(self):
         self._memory_location = 'device'
```

Comment:
Can `_memory_location` ever by anything else? If not it could be a class attribute

Reply:
Yeah seems about right it can be an attribute.

Reply:
Done in [17180eb](https://github.com/pyccel/pyccel/pull/1300/commits/17180ebcb81a2cce22594b086ea3abcd588e45c9).